src/main/java/com/google/api/codegen/transformer/ruby/RubySurfaceNamer.java

```diff
@@ -135,7 +138,10 @@ public String getParamTypeName(ModelTypeTable typeTable, TypeRef type) {
   }
 
   /** The type name for the message property */
-  public String getMessagePropertyTypeName(ModelTypeTable typeTable, TypeRef type) {
+  @Override
+  public String getMessagePropertyTypeName(ImportTypeTable importTypeTable, FieldType fieldType) {
+    ModelTypeTable typeTable = (ModelTypeTable) importTypeTable;
```

Comment:
Is there a way around this casting? This code isn't at a level that should be aware of Gapic vs Discovery. If that would require a bunch of extra work, add a TO DO. 

(It looks like there are multiple sites of casting like this)